### PR TITLE
Increase test coverage

### LINKED ISSUE #149: Increase unit-test coverage
**What happened**:

We want at least a 40% coverage for all our plugins (both frontend and backend). At the moment, the coverage for frontend code is 35.25%:

```
▶ yarn run jest --coverage
--------------------|---------|----------|---------|---------|-------------------------
File                | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s       
--------------------|---------|----------|---------|---------|-------------------------
All files           |   35.25 |    27.44 |   50.67 |   33.72 |                         
```

Note: the backend coverage is 64.4% so that is fine enough.

### PR DESCRIPTION
**Pull request by andresmgot**
Fixes #149 

- Added tests to `appendFrames` (and discovered a potential bug?)
- Added tests to `MetadataInspector` (and migrated to a function component)
- Removed unused component `Forms`

The coverage is now 47% which is not great but at least covers the 40% level that we wanted:

```
File                | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s       
--------------------|---------|----------|---------|---------|-------------------------
All files           |   47.78 |    33.58 |    58.9 |   46.61 |                         
 src                |   26.25 |     8.38 |   21.88 |    26.5 |                         
  DataSource.ts     |   17.86 |     6.04 |   22.73 |   18.25 | 34-55,75,95-326,333-340 
  appendFrames.ts   |   80.95 |       60 |     100 |   84.62 | 8,21,53,58-60           
  module.ts         |       0 |      100 |     100 |       0 | 1-8                     
  regions.ts        |     100 |      100 |     100 |     100 |                         
  requestLooper.ts  |     7.5 |        0 |       0 |     7.5 | 38-111                  
  types.ts          |       0 |        0 |       0 |       0 | 10-23                   
 src/components     |   90.83 |    86.05 |    87.8 |   91.43 |                         
  ConfigEditor.tsx  |   86.11 |    78.57 |   84.62 |   83.87 | 22-30,55-56             
  MetaInspector.tsx |   91.67 |    83.33 |     100 |   91.67 | 15                      
  QueryEditor.tsx   |      98 |     91.3 |   94.44 |   97.83 | 183                     
  Suggestions.tsx   |     100 |      100 |     100 |     100 |                         
  index.ts          |       0 |      100 |       0 |       0 | 1-2                     
  samples.ts        |     100 |      100 |     100 |     100 |                         
  selectors.ts      |     100 |      100 |     100 |     100 |                         
--------------------|---------|----------|---------|---------|-------------------------
```

### CODE REVIEW

File: src/appendFrames.test.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  get: () => 1,
+  toArray: () => [1],
+};
+
+describe('appendMatchingFrames', () => {
+  it('should do nothing with empty input', () => {
+    expect(appendMatchingFrames([], [])).toEqual([]);
+  });
+
+  it('should return input MutableDataFrames', () => {
```
Comment (sarahzinger) [inline src/appendFrames.test.ts:16]: kind of a nit that we can ignore if we don't have good ideas for it:
- in reading this test and the one below I'm not sure I really understand the difference between these two tests. I wonder if we can update the "it" statements to better describe what we're handling or something. Or if these some data frames are mutable should we change them and see that append still handles them the same or something?

File: src/appendFrames.ts [hunk 2]
```diff
@@ -58,6 +58,8 @@ export function appendMatchingFrames(prev: DataFrame[], b: DataFrame[]): DataFra
       for (let i = 0; i < f.length; i++) {
         for (let idx = 0; idx < old.fields.length; idx++) {
           old.fields[idx].values.add(f.fields[idx].values.get(i));
+          // FIXME: I assume this is meant to modify `out` but it's
+          // not doing anything atm
```
Comment (sarahzinger) [inline src/appendFrames.ts:62]: lol should we make a ticket or something?

**Review by sarahzinger — APPROVED** [review]
Looks good to me!

File: src/appendFrames.test.ts (hunk 1 above)
Comment (yaelleC) [reply src/appendFrames.test.ts:16]: What about :

```suggestion
  it('should return input when input is a MutableDataFrames array, () => {
```

and next suggestion?

File: src/appendFrames.test.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    const frame = new MutableDataFrame({ fields: [{ name: 'foo', values: v }] });
+    expect(appendMatchingFrames([frame], [])).toEqual([frame]);
+  });
+
+  it('should return input DataFrame', () => {
```
Comment (yaelleC) [inline src/appendFrames.test.ts:21]:
```suggestion
  it('should return a MutableDataFrame version of input when input is a DataFrames array', () => {
```

File: src/appendFrames.test.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        },
+      ],
+    };
+    const expected = new MutableDataFrame({ fields: [{ name: 'foo', values: v }] });
+    expect(appendMatchingFrames([frame], [])).toEqual([expected]);
+  });
+
+  it('should append new frames', () => {
+    const frame = new MutableDataFrame({ fields: [{ name: 'foo', values: v }] });
+    expect(appendMatchingFrames([], [frame])).toEqual([frame]);
```
Comment (yaelleC) [inline src/appendFrames.test.ts:39]: Should we have a test for two not empty arrays?
_Not sure the following works ;)_

```suggestion
    expect(appendMatchingFrames([frame], [frame])).toEqual([frame, frame]);
```

File: src/components/MetaInspector.tsx [hunk 5]
```diff
@@ -1,15 +1,15 @@
...
+import React from 'react';
 
 import { DataSource } from '../DataSource';
 import { TimestreamOptions, TimestreamQuery } from '../types';
 
 export type Props = MetadataInspectorProps<DataSource, TimestreamQuery, TimestreamOptions>;
 
-export class MetaInspector extends PureComponent<Props> {
-  state = { index: 0 };
+export function MetaInspector(props: Props) {
```
Comment (yaelleC) [inline src/components/MetaInspector.tsx:9]: 👍

File: src/appendFrames.test.ts (hunk 1 above)
Comment (andresmgot) [reply src/appendFrames.test.ts:16]: sounds good to me

File: src/appendFrames.ts (hunk 2 above)
Comment (andresmgot) [reply src/appendFrames.ts:62]: double checking this it actually works :exploding_head: 

I assume that `old.fields[idx]` is pointing to the same vector reference than `out` so modifying that modifies the result. I have added a test that covers this.

**Review by yaelleC — APPROVED** [review]

**Pull request merged**